artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -15,16 +18,300 @@
+                        self._cache_hits += 1
+                        self.logger.debug(f"Cache HIT: NVR {nvr} in group {group}")
+                        return build
+
+            # Otherwise search all groups
+            else:
+                for group_name, group_cache in self._groups.items():
+                    build = group_cache['by_nvr'].get(nvr)
+                    if build:
+                        self._cache_hits += 1
+                        self.logger.debug(f"Cache HIT: NVR {nvr} in group {group_name}")
+                        return build
+
+            self._cache_misses += 1
+            self.logger.debug(f"Cache MISS: NVR {nvr}")
+            return None
+
+    def get_by_name(
+        self,
+        name: str,
+        group: str,
+        outcome: typing.Optional[KonfluxBuildOutcome] = None,
+        assembly: typing.Optional[str] = None,
+        el_target: typing.Optional[str] = None,
+        artifact_type: typing.Optional[ArtifactType] = None,
+        engine: typing.Optional[Engine] = None,
+        embargoed: typing.Optional[bool] = None,
+    ) -> typing.Optional[KonfluxRecord]:
+        """
+        Get latest build for name with optional filters from specified group.
+
+        Returns the most recent build matching all specified criteria.
+
+        :param name: Component name
+        :param group: Group name (required)
+        :param outcome: Filter by outcome (success/failure)
+        :param assembly: Filter by assembly
+        :param el_target: Filter by el_target (e.g., 'el8', 'el9')
+        :param artifact_type: Filter by artifact type (rpm/image)
+        :param engine: Filter by engine (brew/konflux)
+        :param embargoed: Filter by embargoed status
+        :return: Latest matching build or None
+        """
+        with self._lock:
+            # Check if group cached
+            if group not in self._groups:
+                self._cache_misses += 1
+                self.logger.debug(f"Cache MISS: Group {group} not cached")
+                return None
+
+            group_cache = self._groups[group]
+            builds = group_cache['by_name'].get(name, [])
+            if not builds:
+                self._cache_misses += 1
+                self.logger.debug(f"Cache MISS: No builds for name {name} in group {group}")
+                return None
+
+            # Filter builds by criteria
+            for build in builds:  # Already sorted newest first
+                # Apply filters
+                if outcome is not None and build.outcome != outcome:
+                    continue
+                if assembly is not None and build.assembly != assembly:
+                    continue
+                if el_target is not None and build.el_target != el_target:
+                    continue
+                if artifact_type is not None and build.artifact_type != artifact_type:
+                    continue
+                if engine is not None and build.engine != engine:
```

Comment:
I tested 
```
doozer --assembly=stream --group=openshift-4.21 --build-system=konflux --images=agent-installer-ui beta:config:konflux:scan-sources
```
and hit a cache miss in `get_by_name()`. The reason is that `engine` comes from `runtime.build_system` which is a string so the condition `build.engine != engine` is always met. We probably need to add a conversion when entering that function:
```
engine = Engine(engine) if isinstance(engine, str) else engine
```